source/pervasive/result.rs

```diff
@@ -1,9 +1,79 @@
 #[allow(unused_imports)]
+use crate::pervasive::*;
+#[allow(unused_imports)]
 use builtin::*;
 use builtin_macros::*;
 
+verus! {
+
 #[is_variant]
 pub enum Result<T, E> {
     Ok(T),
     Err(E)
 }
+
+impl<T, E> Result<T, E> {
+    #[inline(always)]
+    pub const fn is_ok(&self) -> (res: bool)
+        ensures res <==> self.is_Ok(),
+    {
+        match self {
+            Result::Ok(_) => true,
+            Result::Err(_) => false,
+        }
+    }
+
+    #[inline(always)]
+    pub const fn is_err(&self) -> (res: bool)
+        ensures res <==> self.is_Err(),
+    {
+        match self {
+            Result::Ok(_) => false,
+            Result::Err(_) => true,
+        }
+    }
+
+    // A more-readable synonym for get_Ok_0().
+    #[verifier(inline)]
+    pub open spec fn spec_unwrap_ok(self) -> T
+    recommends self.is_Ok()
+    {
+        self.get_Ok_0()
+    }
+
+    #[verifier(when_used_as_spec(spec_unwrap_ok))]
+    pub fn unwrap_ok(self) -> (t: T)
```

Comment:
Can you please rename this to `unwrap`, to be consistent with https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap ?

Reply:
Updated